subsys/bluetooth/mesh/provisioner.c

```diff
@@ -0,0 +1,746 @@
+/*  Bluetooth Mesh */
+
+/*
+ * Copyright (c) 2017 Intel Corporation
+ * Copyright (c) 2020 Lingao Meng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <errno.h>
+#include <sys/atomic.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/mesh.h>
+#include <bluetooth/uuid.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_PROVISIONER)
+#define LOG_MODULE_NAME bt_mesh_provisioner
+#include "common/log.h"
+
+#include "host/ecc.h"
+#include "host/testing.h"
+
+#include "crypto.h"
+#include "adv.h"
+#include "mesh.h"
+#include "net.h"
+#include "rpl.h"
+#include "beacon.h"
+#include "access.h"
+#include "foundation.h"
+#include "proxy.h"
+#include "prov.h"
+#include "settings.h"
+#include "prov_bearer.h"
+
+static struct {
+	struct bt_mesh_cdb_node *node;
+	uint16_t addr;
+	uint16_t net_idx;
+	uint8_t attention_duration;
+	uint8_t uuid[16];
+	uint8_t method;
+	uint8_t action;
+	uint8_t size;
+} provisioner;
```

Comment:
I think this should be renamed to something like `provisioning_state`. Simply `provisioner` is quite vague... (I'm to blame for that :p)